### PR TITLE
datetime-local: default value

### LINKED ISSUE #8846: When selecting the current date (for type instant), the web component should add a default time
##### **Overview of the feature request**

When you have a date of type `Instant`, the Angular update component uses a `type="datetime-local"`. When selecting the current day on the HTML component, only the date is set, not the time. So you have to add it manually otherwise the component does not allow you to save it.

It would be helpful to set the time (by default could be 00:00)

##### **Motivation for or Use Case**

Imagine you have the following JDL which represents an item status with a status date of type `Instant`:

```
entity ItemStatus{
    status String required,
    statusDate Instant required,
    additionalInformation String
}
```

The code generated in the `item-status-update.component.html` uses a 

```
<div class="d-flex">
    <input id="field_statusDate" type="datetime-local" class="form-control" name="statusDate" [(ngModel)]="statusDate"
    required/>
</div>
```

So when you create a new Item Status and you hit the current day button, only the date is set, not the time. Because the date in not valid, the Save button is not enabled, and you can't save the Item Status:

instant1

You have to manually set a time for the Save button be enabled. The component should add 00:00 by default.

instant2

##### **Related issues or PR**

See #5006
See #6711

-   [X] Checking this box is mandatory (this is just to show you read everything)

### PR DESCRIPTION
**Pull request by qmonmert**
Fix #8846

Set datetime-local by default with the date: today 00:00

-   [x] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
-   [x] Tests are added where necessary
-   [x] Documentation is added/updated where necessary
-   [x] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/master/CONTRIBUTING.md) are followed

### CODE REVIEW

File: generators/entity-client/templates/angular/src/main/webapp/app/entities/entity-management-update.component.ts.ejs [hunk 1]
```diff
@@ -167,6 +167,13 @@ for ( const idx in variables ) { %>
             <%- queries[idx] %>
 <%_ } _%>
         });
+        <%_ for ( idx in fields ) {
```
Comment (MathieuAA) [inline generators/entity-client/templates/angular/src/main/webapp/app/entities/entity-management-update.component.ts.ejs]: For consistency, please remove the extra spaces around the iteration instruction inside the for

File: generators/entity-client/templates/angular/src/main/webapp/app/entities/entity-management-update.component.ts.ejs [hunk 2]
```diff
@@ -167,6 +167,13 @@ for ( const idx in variables ) { %>
...
+        const fieldName = fields[idx].fieldName;
+        const fieldType = fields[idx].fieldType;
+            if (['Instant', 'ZonedDateTime'].includes(fieldType)) { _%>
```
Comment (MathieuAA) [inline generators/entity-client/templates/angular/src/main/webapp/app/entities/entity-management-update.component.ts.ejs]: I don't understand the indentation here

File: generators/entity-client/templates/angular/src/main/webapp/app/entities/entity-management-update.component.ts.ejs [hunk 3]
```diff
@@ -286,6 +293,20 @@ _%>
 <%_ } _%>
         };
     }
+    <%_ for ( idx in fields ) {
```
Comment (MathieuAA) [inline generators/entity-client/templates/angular/src/main/webapp/app/entities/entity-management-update.component.ts.ejs]: For consistency, please remove the extra spaces around the iteration instruction inside the for

File: generators/entity-client/templates/angular/src/main/webapp/app/entities/entity-management-update.component.ts.ejs [hunk 4]
```diff
@@ -286,6 +293,20 @@ _%>
...
+        const fieldName = fields[idx].fieldName;
+        const fieldType = fields[idx].fieldType;
+            if (['Instant', 'ZonedDateTime'].includes(fieldType)) { _%>
```
Comment (MathieuAA) [inline generators/entity-client/templates/angular/src/main/webapp/app/entities/entity-management-update.component.ts.ejs]: the same, I don't get the indentation

File: generators/entity-client/templates/angular/src/main/webapp/app/entities/entity-management-update.component.ts.ejs [hunk 5]
```diff
@@ -286,6 +293,20 @@ _%>
...
+
+    private <%= fieldName %>SetDefaultValue(): void {
+        const field = this.editForm.get('<%= fieldName %>');
+        if (field && field.value === null) {
+            const t = moment().startOf('day').format(DATE_TIME_FORMAT);
```
Comment (MathieuAA) [inline generators/entity-client/templates/angular/src/main/webapp/app/entities/entity-management-update.component.ts.ejs]: please use a better name than "t"

File: generators/client/templates/react/src/main/webapp/app/shared/util/date-utils.ts.ejs [hunk 6]
```diff
@@ -25,3 +25,5 @@ export const convertDateTimeFromServer = date =>
 
 export const convertDateTimeToServer = date =>
   (date ? moment(date, APP_LOCAL_DATETIME_FORMAT_Z).toDate() : null);
+
+export const defaultDateTime = () => moment().startOf('day').format(APP_LOCAL_DATETIME_FORMAT);
```
Comment (MathieuAA) [inline generators/client/templates/react/src/main/webapp/app/shared/util/date-utils.ts.ejs]: it's a function, please use a verb to mark the action (like convertXXX, whatever).

**Comment by clementdessoude** [comment]
Won't this mess with the required argument? If the date is required and the user do not select a date, will it throw an error?

File: generators/client/templates/react/src/main/webapp/app/shared/util/date-utils.ts.ejs (hunk 6 above)
Comment (qmonmert) [reply generators/client/templates/react/src/main/webapp/app/shared/util/date-utils.ts.ejs]: ok

File: generators/entity-client/templates/angular/src/main/webapp/app/entities/entity-management-update.component.ts.ejs (hunk 1 above)
Comment (qmonmert) [reply generators/entity-client/templates/angular/src/main/webapp/app/entities/entity-management-update.component.ts.ejs]: ok

File: generators/entity-client/templates/angular/src/main/webapp/app/entities/entity-management-update.component.ts.ejs (hunk 3 above)
Comment (qmonmert) [reply generators/entity-client/templates/angular/src/main/webapp/app/entities/entity-management-update.component.ts.ejs]: ok

File: generators/entity-client/templates/angular/src/main/webapp/app/entities/entity-management-update.component.ts.ejs (hunk 5 above)
Comment (qmonmert) [reply generators/entity-client/templates/angular/src/main/webapp/app/entities/entity-management-update.component.ts.ejs]: ok "today" will be better

File: generators/entity-client/templates/angular/src/main/webapp/app/entities/entity-management-update.component.ts.ejs (hunk 2 above)
Comment (qmonmert) [reply generators/entity-client/templates/angular/src/main/webapp/app/entities/entity-management-update.component.ts.ejs]: I changed that

File: generators/entity-client/templates/angular/src/main/webapp/app/entities/entity-management-update.component.ts.ejs (hunk 4 above)
Comment (qmonmert) [reply generators/entity-client/templates/angular/src/main/webapp/app/entities/entity-management-update.component.ts.ejs]: I changed that

**Comment by qmonmert** [comment]
@clementdessoude if the field is required and the user don't select a date, there is an error message like before 
demo

**Comment by qmonmert** [comment]
@MathieuAA are you ok with my updates ?

**Comment by MathieuAA** [comment]
Thanks @qmonmert

**Comment by qmonmert** [comment]
@MathieuAA :)

**Pull request merged**